### DISCUSSION TITLE
Be Able to Assign Category When Saving New Automation

### DISCUSSION THREAD

**Discussion by gmoney63** [home-assistant/frontend#21782] [Automations / scripts / scenes] [answered]
As of HA 2024.8.2

I would like to be able to assign a category WHILE naming/saving a new automation. 

The current process is: you create an automation, click Save, name the new automation, go back to the automations list, find your new automation in the "Ungrouped" section, THEN you can assign a category.

_Please don't mistake this feature request as unappreciative for the organizational work that has been done with automations._

The picture below shows the pop-up for saving a new automation which is where I'd love to see a drop down for assigning a category:

![image](https://github.com/user-attachments/assets/93d9153c-42f5-43f2-8d11-da5b73adcb7c)

**Comment by ildar170975** [comment]
From another side, many users do not use categories or labels - then showing theme these fields may cause irritation & confusion (unless these options are hidden inside some collapsed panel).
Also, same FR may be applied for labels: if some user uses labels extensively - he/she would like to add a labels in the same dialog.

**Reply by gmoney63** [reply]
Agreed but I think if a user doesn't use categories then it could default to "Ungrouped" just as it currently does. 

I just feel if we're able to assign "Areas" when adding to devices, Labels to devices, entities, etc.. then we should be able to assign a category to a new automation as well.

This isn't a deal-breaking feature that would make me stop using HA lol it's more for ease in cutting out a few extra steps. If it's determined that this feature would cause more harm than good, then that's completely okay and I'll continue to do it the way that it currently is.

**Reply by ildar170975** [reply]
> Agreed but I think if a user doesn't use categories then it could default to "Ungrouped" just as it currently does.

Ofc, my doubts were only about "do not show a user fields which he never uses".
For a beginner user - this might be a confusion; for experienced users who do not use categories - this will be an irritation.
As I suggested, these extra fields like "category", "label" may be shown in a collapsed panel (this is a common way of showing options which are not frequently used).

**Reply by JelleKoster** [reply]
@ildar170975 Whether it should be shown as collapsed by default should not depend on whether it is frequently used or not.
It should be based on the vision whether it is best practice to use it.
Currently, new users might not encounter the feature and get bogged down with a large unstructured list at some point.

If it defaults to Ungrouped (therefore not complaining about a mandatory field needing to be filled) that should be fine for new users / users who don't use the functionality.

I was about to raise this same FR, have my upvote =)

**Comment by distante** [comment]
I was about to post this exact same thing. I also would be happy to have this option.

**Comment by silamon** [comment, marked as answer]
https://www.home-assistant.io/blog/2025/01/03/release-20251/#setting-categories-and-labels-when-creating-automations-and-scripts
